lisa/tools/python.py

```diff
@@ -89,3 +94,79 @@ def exists_package(self, package_name: str) -> bool:
     def uninstall_package(self, package_name: str) -> bool:
         result = self.run(f"uninstall {package_name} -y", force_run=True, sudo=True)
         return result.exit_code == 0
+
+
+class PythonVenv(Tool):
+    @property
+    def command(self) -> str:
+        return f"{self.get_venv_path()}/bin/{self.python.command}"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Python]
+
+    def _create_venv(self, venv_path: str = "", sudo: bool = False) -> PurePath:
```

Comment:
To make sure no side effects among test cases, make `venv_path` as a required argument. So the path of test case should be passed in. Or if some test cases share the same env, they can use the same string. But the `"venv"` is not safe as a default.

Reply:
When the `sudo` is needed? The venv should be a user path always.

Reply:
`_venv_path: PurePath = (
            PurePath(venv_path)
            if venv_path
            else self.node.working_path / datetime.utcnow().isoformat()
)`

Will this work? Cases that want to reuse the env can pass venv_path, otherwise the default would always be unique.

Reply:
The timestamp is not an idempotent way. If some cases want to share a path, you can hardcode a path name in a common logic of those test cases.

Reply:
Removed default path